### ISSUE TITLE
Guest auth fails when specifying an incorrect password and forcing smb2

### ISSUE THREAD

**Issue by pfsen** [jfjallid/go-shareenum#4]
```
user@box:~/tools$ ./go-shareenum -host 10.10.11.175 -enum -user 'user' -smb2
Enter password: 
2023/07/06 13:46:20 [Notice] [-] Signing is required
2023/07/06 13:46:20 [Notice] [+] Login successful
2023/07/06 13:46:20 connection.go:192: smb [Error] Skip: Signing is required and invalid signature found
```

**Comment by jfjallid** [comment]
Seems to be a problem with the go-smb library interpreting the "signature required" attribute too strictly even when authenticated as a guest.

**Comment by jfjallid** [comment]
Fixed in commit 948f327ee299058cc730552b2adc996dd4e59f59
